bastionzero/service/targets/database.go

```diff
@@ -14,6 +14,26 @@ const (
 	databaseSinglePath = databaseBasePath + "/%s"
 )
 
+// CreateDatabaseTargetRequest is used to create a new Database target
+type CreateDatabaseTargetRequest struct {
+	TargetName      string `json:"targetName,omitempty"`
+	ProxyTargetID   string `json:"proxyTargetId,omitempty"`
+	RemoteHost      string `json:"remoteHost,omitempty"`
+	RemotePort      *Port  `json:"remotePort,omitempty"`
+	LocalPort       *Port  `json:"localPort,omitempty"`
+	LocalHost       string `json:"localHost,omitempty"`
+	IsSplitCert     bool   `json:"splitCert,omitempty"`
+	DatabaseType    string `json:"databaseType,omitempty"`
+	EnvironmentID   string `json:"environmentId,omitempty"`
+	EnvironmentName string `json:"environmentName,omitempty"`
+}
```

Comment:
For struct types that are only used in the _create_ endpoint methods, I'm trying to only include `omitempty` if the field is optional as defined by the API spec. Additionally, I make the field a pointer type if the default/zero value has semantic meaning that is different than null/unspecified.

So with that in mind, can you make the following changes?

- Remove `omitempty` from `TargetName`, `ProxyTargetID`, `RemoteHost`, and `RemotePort` since these fields are required
- Remove the pointer type for `RemotePort`: `RemotePort *Port` --> `RemotePort Port`

Reply:
@ymarcus93, should I remove the pointer type from `LocalPort` as well? Looking at https://cloud.bastionzero.com/api/#get-/api/v2/targets/database/-id-, doesn't seem like the zero value has a semantic meaning.